1.2.2.x/input form save

sharrief opened the pull request:
## This PR is built on top of #49 and #48. 
Those should be merged first as this PR includes those changes.

## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Fixes remaining issues to enable the E2E input form save / new service request/ticket scenario. 

## Related Tickets & Documents

[1.2 User creates request](https://projects.digitalaidseattle.org/features-details?recordId=recFJrAV8KeddrkLu)


## QA Instructions, Screenshots, Recordings

1. Install, build and run the client: 'yarn install && yarn dev`
1. Click new request
1. Fill in the all the information
1. Click Save
1. Verify form closes 
1. Check supabase for newly created records

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
- [ ] Semantic HTML implemented?
- [ ] Keyboard operability supported?
- [ ] Color contrast tested?

## Added/updated tests?
_We are aiming for 100% statement coverage._

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [optional] What gif best describes this PR or how it makes you feel?

![whoa, we're halfway there](https://media1.tenor.com/m/Cuf0ngxB-egAAAAC/halfway-there-bon-jovi.gif)

DillonOLeary reviewed (commented):
<img width="863" alt="Screenshot 2024-05-12 at 5 41 51 PM" src="https://github.com/digitalaidseattle/seattle-humane-app/assets/25556678/1d5101e4-b7b0-4021-93fc-e713882be51f">
Seems good, but I followed the steps and got the error above. Is the issue that client service still uses `phone` instead of `phone_number`? https://github.com/digitalaidseattle/seattle-humane-app/pull/50/files#diff-a1cbcb775eb6d4fde0f6000a69f165e005991617218f4a93aa37250876c85646R168

sharrief commented on .vscode/launch.json:
```diff
@@ -15,13 +15,21 @@
-      "type": "node-terminal",
+      "type": "node",
       "request": "launch",
-      "command": "yarn dev",
```
Sorry, I'll look into these a bit more. I just copied the default settings from NextJS because the existing setting were not working for me. I could not get breakpoints attach

sharrief replied:
Reverted this change

sharrief commented on src/components/serviceRequest/ServiceRequestDialog.tsx:
```diff
@@ -58,11 +59,12 @@ function ServiceRequestDialog(props: ServiceRequestDialogProps) {
   ] = useReducer(serviceInfoReducer, defaultServiceInformation);
 
-  const cancel = () => {
+  const close = () => {
```
This code is about resetting the state and closing the dialog, which is what we do after a cancel but now we also do it after a save. So renaming it to something that applied to both circumstances made more sense

DillonOLeary replied:
Makes sense!

sharrief commented on .vscode/launch.json:
```diff
@@ -15,13 +15,21 @@
+        "<node_internals>/**"
+      ],
       "serverReadyAction": {
+        "action": "debugWithEdge",
```
Reverted this change

sharrief commented:
> Seems good, but I followed the steps and got the error above. Is the issue that client service still uses `phone` instead of `phone_number`? https://github.com/[/pull/50](https://github.com/digitalaidseattle/seattle-humane-app/pull/50)/files#diff-a1cbcb775eb6d4fde0f6000a69f165e005991617218f4a93aa37250876c85646R168

I must have had old generated types in my branch. I fixed the issue and added an explicit type to the field list to avoid a de-sync in the future.
